src/utils/timeFromNow.jsx

```diff
@@ -0,0 +1,127 @@
+    relativeTime: PropTypes.shape({
+        s: PropTypes.string,
+        ss: PropTypes.string,
+        m: PropTypes.string,
+        mm: PropTypes.string,
+        h: PropTypes.string,
+        hh: PropTypes.string,
+        d: PropTypes.string,
+        dd: PropTypes.string,
+        M: PropTypes.string,
+        MM: PropTypes.string,
+        y: PropTypes.string,
+        yy: PropTypes.string,
+    }),
+    relativeTimeRounding: PropTypes.oneOf([Math.round, Math.floor]),
+    relativeTimeThreshold: PropTypes.shape({
+        s: PropTypes.number,
+        m: PropTypes.number,
+        h: PropTypes.number,
+        d: PropTypes.number,
+        M: PropTypes.number,
+    }),
+};
+
+const defaultProps = {
+    className: undefined,
+    id: undefined,
+    locale: 'en',
+    style: {},
+    relativeTime: {},
+    relativeTimeRounding: undefined,
+    relativeTimeThreshold: {},
+};
+
+function TimeFromNow(props) {
+    const {
+        className,
+        date,
+        id,
+        locale,
+        style,
+        relativeTime,
+        relativeTimeThreshold,
+        relativeTimeRounding,
+    } = props;
+    const containerClasses = ClassNames('ui', 'time_from_now', className);
+    const defaultRelativeTime = moment().locale(locale).localeData()._relativeTime; // eslint-disable-line no-underscore-dangle, max-len
+    const defaultRounding = moment.relativeTimeRounding();
+    /* eslint-disable camelcase */
+    const defaultThreshold_ss = moment.relativeTimeThreshold('ss');
+    const defaultThreshold_s = moment.relativeTimeThreshold('s');
+    const defaultThreshold_m = moment.relativeTimeThreshold('m');
+    const defaultThreshold_h = moment.relativeTimeThreshold('h');
+    const defaultThreshold_d = moment.relativeTimeThreshold('d');
+    const defaultThreshold_M = moment.relativeTimeThreshold('M');
+    /* eslint-enable camelcase */
+
+    // Customizing moment's relativeTime
+    if (relativeTime) {
+        moment.updateLocale(locale, { relativeTime });
+    }
+
+    // Customizing moment's relativeTimeRounding
+    if (relativeTimeRounding) {
+        moment.relativeTimeRounding(relativeTimeRounding);
+    }
+
+    // Customizing moment's relativeTimeThreshold
+    if (relativeTimeThreshold) {
+        /* eslint-disable camelcase */
+        moment.relativeTimeThreshold('ss', relativeTimeThreshold.ss || defaultThreshold_ss);
+        moment.relativeTimeThreshold('s', relativeTimeThreshold.s || defaultThreshold_s);
+        moment.relativeTimeThreshold('m', relativeTimeThreshold.m || defaultThreshold_m);
+        moment.relativeTimeThreshold('h', relativeTimeThreshold.h || defaultThreshold_h);
+        moment.relativeTimeThreshold('d', relativeTimeThreshold.d || defaultThreshold_d);
+        moment.relativeTimeThreshold('M', relativeTimeThreshold.M || defaultThreshold_M);
+        /* eslint-enable camelcase */
+    }
+
+    // Setting fromNow string with customized relativeTime
+    const fromNowTime = moment(date).fromNow(true);
```

Comment:
Are we presuming that the `date` value has already been shifted to the user's present time zone?
I'm pretty sure `moment(date)`, if `date` is a string representing a date/time, and does not have timezone or UTC offset, will instantiate a moment in the current time zone of the machine it's executing on.  That's a recipe for possible time zone troubles.

This is not necessarily a problem; I'm just flagging it as a potential problem.  We'd want to do some testing to be sure.

Reply:
Same here, as above, hopefully it'll just be solved by the propTypes though.

Reply:
Yeah, I think let's take a wait and see approach here.  If it ain't broke, don't fix it.  If we start having time zone blues, we'll dig in and see what's what.